cli/popper/gha.py

```diff
@@ -156,47 +156,57 @@ def download_actions(self):
             if not os.path.exists(repo_parent_dir):
                 os.makedirs(repo_parent_dir)
 
-            if not infoed:
+            if not dry_run and not infoed:
                 pu.info('[popper] cloning actions from repositories\n')
                 infoed = True
 
-            scm.clone(user, repo, repo_parent_dir, version)
+                scm.clone(user, repo, repo_parent_dir, version)
 
-            cloned.add('{}/{}'.format(user, repo))
+                cloned.add('{}/{}'.format(user, repo))
 
-    def instantiate_runners(self, verbose=False):
+    def instantiate_runners(self, verbose=False, dry_run=False):
         """Factory of ActionRunner instances, one for each action"""
         for _, a in self.wf['action'].items():
-            if 'docker://' in a['uses']:
-                a['runner'] = DockerRunner(a, self.workspace,
-                                           self.env, self.timeout, verbose)
-                continue
-
-            if './' in a['uses']:
-                if os.path.exists(os.path.join(a['uses'], 'Dockerfile')):
+            if not dry_run:
```

Comment:
It might be cleaner to define a global `msg_prefix` (or `dryrun_msg`, or something alike). This string is always included in the messages and the value of the string is assigned based on whether users requested a dry run. In this way we only need to check for a dry run right before an `exec_cmd` invocation, like:

```python
if not dry_run:
    pu.exec_cmd(...)
```

does that make sense?

Reply:
@ivotron instead of putting `if` checks before `pu.exec_cmd(...)`, I inserted one over here. Never actually executing `exec_cmd`, a true dry run if you will. https://github.com/systemslab/popper/blob/f98654f4074bbed8b43927492bebd0f87b5f975a/cli/popper/utils.py#L147-L149

Reply:
@ivotron added `msg_prefix` in the `ActionRunner` class so that it may be used by any `Runner` instance if required.
https://github.com/systemslab/popper/blob/f98654f4074bbed8b43927492bebd0f87b5f975a/cli/popper/gha.py#L288-L299